static/app/stores/Commiters/CommittersContext.tsx

```diff
@@ -0,0 +1,279 @@
+import {createContext, useContext, useEffect, useReducer} from 'react';
+
+import {Client} from 'sentry/api';
+import {
+  AvatarProject,
+  Committer,
+  Event,
+  Group,
+  Organization,
+  Project,
+} from 'sentry/types';
+import getDisplayName from 'sentry/utils/getDisplayName';
+import useApi from 'sentry/utils/useApi';
+
+function exhaustive(x?: never) {
+  throw new Error(
+    `Unhandled ${JSON.stringify(x)} switch case action in committersReducer`
+  );
+}
+
+function makeCommitterStoreKey({
+  organizationSlug,
+  projectSlug,
+  eventId,
+}: {
+  eventId: string;
+  organizationSlug: string;
+  projectSlug: string;
+}): string {
+  return `${organizationSlug} ${projectSlug} ${eventId}`;
+}
+
+export function fetchCommitters(
+  api: Client,
+  params: {eventId: string; organizationSlug: string; projectSlug: string}
+): Promise<{committers: Committer[]}> {
+  return api.requestPromise(
+    `/projects/${params.organizationSlug}/${params.projectSlug}/events/${params.eventId}/committers/`,
+    {
+      method: 'GET',
+    }
+  );
+}
+
+type ResetAction = {
+  type: 'reset';
+};
+
+type SetCommittersError = {
+  payload: {
+    eventId: string;
+    organizationSlug: string;
+    projectSlug: string;
+  };
+  type: 'set error';
+};
+
+type StartLoadingCommitters = {
+  payload: {
+    eventId: string;
+    organizationSlug: string;
+    projectSlug: string;
+  };
+  type: 'start loading';
+};
+
+type AddCommitters = {
+  payload: {
+    committers: Committer[];
+    eventId: string;
+    organizationSlug: string;
+    projectSlug: string;
+  };
+  type: 'add committers';
+};
+
+export type CommittersState = Record<
+  string,
+  {
+    committers: Committer[];
+    committersError: boolean;
+    committersLoading: boolean;
+  }
+>;
+
+export type CommittersAction =
+  | ResetAction
+  | AddCommitters
+  | StartLoadingCommitters
+  | SetCommittersError;
+
+export function committersReducer(state, action: CommittersAction): CommittersState {
+  switch (action.type) {
+    case 'add committers': {
```

Comment:
should these strings be in an `enum`?

Reply:
No, there is no need. Usually this results in stuff like
```tsx
enum Actions = {
  AddCommitters = "add committers"
}
```
or something similar which is just unnecessary duplication

Reply:
I can still see the benefits of enums :)

Reply:
Maybe if this gets too complicated, we can think of enums, but I would simple and build from there.